packages/verified-fetch/test/verified-fetch.spec.ts

```diff
@@ -523,4 +523,31 @@ describe('@helia/verifed-fetch', () => {
       await expect(resp.text()).to.eventually.equal('hello world')
     })
   })
+
+  describe('custom dns-resolvers', () => {
+    it('uses custom dnsResolvers if provided', async () => {
+      const customResolver1 = Sinon.stub()
+      const customResolver2 = Sinon.stub()
+      const onProgress = Sinon.stub()
+
+      customResolver1.rejects(new Error('Could not resolve PeerId "mydomain.com"'))
+      customResolver2.returns(Promise.resolve('/ipfs/QmVP2ip92jQuMDezVSzQBWDqWFbp9nyCHNQSiciRauPLDg'))
+
+      const verifiedFetch = new VerifiedFetch({
+        helia
+      }, {
+        dnsResolvers: [customResolver1, customResolver2]
+      })
+      try {
+        await verifiedFetch.fetch('ipns://mydomain.com', { onProgress })
+      } catch {
+        // ignoring error of walking the CID because we haven't actually added the block to the blockstore
+      }
```

Comment:
If we expect this to reject, it's worth ensuring it does reject and in the expected way:

```suggestion
      // ignoring error of walking the CID because we haven't actually added the block to the blockstore
      await expect(verifiedFetch.fetch('ipns://mydomain.com', { onProgress })).to.eventually.be.rejected
        .with.property('code', 'ERR_EXPECTED_ERR_CODE')
```